src/Microsoft.AspNetCore.Hosting/Startup/StartupBase.cs

```diff
@@ -16,4 +16,25 @@ public virtual IServiceProvider ConfigureServices(IServiceCollection services)
             return services.BuildServiceProvider();
         }
     }
+
+    public abstract class StartupBase<TContainer> : IStartup
+    {
+        private readonly IServiceProviderFactory _factory;
+
+        public StartupBase(IServiceProviderFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public abstract void Configure(IApplicationBuilder app);
+
+        public IServiceProvider ConfigureServices(IServiceCollection services)
+        {
+            var builder = _factory.CreateBuilder(services);
```

Comment:
If you're supposed to call `Populate` on the builder before returning it from the factory, don't you have to call `ConfigureServices` so services are actually added before the builder's populated?